test/libsolidity/smtCheckerTests/functions/functions_trivial_condition_for.sol

```diff
@@ -5,4 +5,3 @@ contract C
 	function f(bool x) public pure { require(x); for (;x;) {} }
 }
 // ----
-// Warning: (98-99): For loop condition is always true.
```

Comment:
Why is this removed?

Reply:
Because this PR disables the triviality check for loop conditions, since now we don't just run once and erase all knowledge but rather consider the two branches (in loop and skip loop). Also, it is common that the first time a condition is evaluated during execution it is true, which might not be true at some point.

Reply:
But how can we be sure?

Reply:
In this case we know `x` is `true` for the first iteration, but we don't know it's always true without analyzing the body in an unbounded way.

Reply:
OK, I see what you mean. So the error for now had a high false-positive rate? Didn't we at least do something like clear everything that is touched in the loop body?

We should at least be able to infer whether it is always false.

Reply:
Yes, this PR keeps that clearing part, but it also considers two separate branches and merges them.
True, I'll keep the always false check then.